### PR TITLE
Added description header field in backstage.io docs

### LINKED ISSUE #1994: Docs from backstage.io/docs having markdown as description
E.g. https://backstage.io/docs/features/techdocs/techdocs-overview

> TechDocs Documentation · Backstage
> \#\# What is it?

The problem being that .md files in /docs folder of the repo doesn't have `description` document header field (https://docusaurus.io/docs/en/doc-markdown#documents).

So the solution is to go through all files and add `description` for each one of them ;)

### PR DESCRIPTION
**Pull request by iamrajiv**
## Hey, I just made a Pull Request!

Fixes: #1994 

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: docs/dls/figma.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 id: figma
 title: Figma
+description: Documentation on Using Figma to build your own plugins for
+Backstage
```
Comment (shmidt-i) [inline docs/dls/figma.md:5]:
```suggestion
description: Documentation on using Figma to build your own plugins for
Backstage
```

**Comment by iamrajiv** [comment]
@shmidt-i @stefanalund @OrkoHunter Please review.

File: docs/features/techdocs/concepts.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ---
 id: concepts
 title: Concepts
+description: Documentation on concepts that are introduced with Spotify's docs
+like code solution in Backstage
```
Comment (shmidt-i) [inline docs/features/techdocs/concepts.md:5]:
```suggestion
description: Documentation on concepts that are introduced with Spotify's docs-like-code solution in Backstage
```

File: docs/features/techdocs/creating-and-publishing.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 id: creating-and-publishing
 title: Creating and publishing your docs
 sidebar_label: Creating and Publishing Documentation
+description: Guidance on How to Create and Publish Documentation
```
Comment (shmidt-i) [inline docs/features/techdocs/creating-and-publishing.md:5]:
```suggestion
sidebar_label: Creating and Publishing Documentation
description: Guidance on how to create and publish documentation
```

File: docs/plugins/existing-plugins.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 id: existing-plugins
 title: Existing plugins
+description: Lists of open source Existing plugins
```
Comment (shmidt-i) [inline docs/plugins/existing-plugins.md:4]:
```suggestion
description: Lists of existing open source plugins
```

File: docs/plugins/structure-of-a-plugin.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ---
 id: structure-of-a-plugin
 title: Structure of a Plugin
+description: Details about Structure of a Plugin
```
Comment (shmidt-i) [inline docs/plugins/structure-of-a-plugin.md:4]:
```suggestion
description: Details about structure of a plugin
```

File: docs/tutorials/journey.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
 ---
 id: journey
 title: Future developer journey
+description: Document describes a possible journey of a future Backstage
```
Comment (shmidt-i) [inline docs/tutorials/journey.md:4]:
```suggestion
description: This document describes a possible journey of a future Backstage
```

**Review by shmidt-i — APPROVED** [review]
Great, thanks, small nitpicks here and there, but will be happy to merge after!

**Review by OrkoHunter — APPROVED** [review]
Wow, great work @iamrajiv. Really appreciate it.

**Comment by iamrajiv** [comment]
Thank You @OrkoHunter and @shmidt-i

**Pull request merged**